dagshub/common/cli.py

```diff
@@ -52,6 +55,106 @@ def mount(ctx, verbose, **kwargs):
     mount(**kwargs)
 
 
+@cli.command()
+@click.option("--repo_name", help="The repository name to set up")
+@click.option("--repo_owner", help="The organization under which the repository needs to be set up")
+@click.option("--url", help="DagsHub remote url; either provide --url or repo_name and repo_owner")
+@click.option("--host", default=config.DEFAULT_HOST, help="DagsHub instance to which you want to login")
+@click.option("--mlflow", default=True, is_flag=True, help="Set up MLFlow environment with DagsHub remote")
+@click.option("--dvc", default=False, is_flag=True, help="Set up DVC with DagsHub remote")
+@click.pass_context
+def init(ctx, repo_name, repo_owner, url, host, mlflow, dvc):
+    _init(repo_name, repo_owner, url, host, mlflow, dvc)
+
+
+def _init(repo_name=None, repo_owner=None, url=None, root=None,
+          host=config.DEFAULT_HOST, mlflow=True, dvc=False):
+    from ..upload.wrapper import create_repo
+    import urllib
+
+    # Setup required variables
+    root = root or get_project_root(Path(os.path.abspath('.')))
+
+    if url and (repo_name or repo_owner):
+        repo_name, repo_owner = None, None
+
+    if not url:
+        if repo_name is not None and repo_owner is not None:
+            url = urllib.parse.urljoin(host, f'{repo_owner}/{repo_name}')
+        else:
+            with open(root / '.git' / 'config', 'r') as cfg:
+                for line in cfg.readlines():
+                    if host in line:
+                        url = line.split()[-1][:-4]
+                        break
+    elif url[-4] == '.':
+        url = url[:-4]
+
+    if not (repo_name and repo_owner):
+        splitter = lambda x: (x[-1], x[-2]) # noqa E721
+        repo_name, repo_owner = splitter(url.split('/'))
+
+    if None in [repo_name, repo_owner, url]:
+        raise ValueError('Host not found; please specify the remote url with --url')
```

Comment:
This error message doesn't actually reflect what's happening. The user might have specified the --url and put in a wrong URL, and will be confused.
"Host not found" also doesn't reflect what's actually happening here.
Suggestion:
```suggestion
        raise ValueError('Could not parse repository owner and name. Make sure you specify either a link to the repository with --url or a pair of --repo-owner and --repo-name').
```

Also the thing about not scaring the user with stack traces still applies.